packages/block-editor/src/components/link-control/test/link-value-transforms.js

```diff
@@ -0,0 +1,161 @@
+	opensInNewTab: {
+		dataKey: 'linkTarget',
+		toLink: ( value ) => value === '_blank',
+		toData: ( value ) => ( value ? '_blank' : undefined ),
+	},
+	noFollow: {
+		dataKey: 'linkRel',
+		toLink: ( value ) => value.includes( 'nofollow' ),
+		toData: ( value, _, { linkRel: currentLinkRel } ) => {
+			// if the value is truthy and the current value is set
+			// then append otherwise just add the value
+			if ( value && currentLinkRel ) {
+				return `${ currentLinkRel } nofollow`;
+			} else if ( value ) {
+				return 'nofollow';
+			}
+		},
+	},
+	sponsored: {
+		dataKey: 'linkRel',
+		toLink: ( value ) => value.includes( 'sponsored' ),
+		toData: ( value, _, { linkRel: currentLinkRel } ) => {
+			// if the value is truthy and the current value is set
+			// then append otherwise just add the value
+			if ( value && currentLinkRel ) {
+				return `${ currentLinkRel } sponsored`;
+			} else if ( value ) {
+				return 'sponsored';
+			}
+		},
+	},
+};
+
+describe( 'building a link value from data', () => {
+	it.each( [
+		[
+			{
+				href: 'https://www.wordpress.org',
+				postType: 'post',
+				id: 123,
+				linkTarget: '_blank',
+				linkRel: 'nofollow noopenner sponsored',
+				keyToIgnore: 'valueToIgnore',
+			},
+			{
+				url: 'https://www.wordpress.org',
+				type: 'post',
+				id: 123,
+				opensInNewTab: true,
+				noFollow: true,
+				sponsored: true,
+			},
+		],
+		[
+			{
+				href: 'https://www.wordpress.org',
+				postType: 'post',
+				id: 123,
+				linkRel: 'sponsored neyfollow',
+			},
+			{
+				url: 'https://www.wordpress.org',
+				type: 'post',
+				id: 123,
+				opensInNewTab: false,
+				noFollow: false,
+				sponsored: true,
+			},
+		],
+	] )(
+		'build a valid link value from supplied data mapping',
+		( data, expected ) => {
+			const { toLink } = getLinkValueTransforms( mapping );
+
+			const linkValue = toLink( data );
+
+			expect( linkValue ).toEqual( expected );
+		}
+	);
+
+	it( 'returns raw data attribute value when toLink transform is not callable', () => {
+		const { toLink } = getLinkValueTransforms( {
+			url: {
+				dataKey: 'href',
+				// allows toLink to be ommitted in case of simple mapping
+				// but still allows toData to be defined.
+				toData: identity,
+			},
+		} );
+
+		const linkValue = toLink( {
+			href: 'https://www.wordpress.org',
+		} );
+
+		expect( linkValue ).toEqual( {
+			url: 'https://www.wordpress.org',
+		} );
+	} );
+} );
+
+describe( 'building data from a link value', () => {
+	it( 'build a valid data object from supplied link value mapping', () => {
+		const linkValue = {
+			url: 'https://www.wordpress.org',
+			type: 'post',
+			id: 123,
+			opensInNewTab: true,
+			noFollow: true,
+			sponsored: true,
+		};
+
+		const { toData } = getLinkValueTransforms( mapping );
+		const data = toData( linkValue );
+
+		expect( data ).toEqual( {
+			href: 'https://www.wordpress.org',
+			postType: 'post',
+			id: 123,
+			linkTarget: '_blank',
+			linkRel: 'nofollow sponsored',
```

Comment:
@getdave I think the expected linkRel should be `noopener noreferer nofollow sponsored` because the linkValue is
```
opensInNewTab: true,
noFollow: true,
sponsored: true,
```
This test case is passed with incorrect value...

Reply:
The failed test is found when running unit test against [link-value-transforms-with-utils.js](https://github.com/WordPress/gutenberg/pull/55143/files#diff-e5e501762312c7f48f908cb82c72913a5dd612780e669fe88cb999ccf4a26879)

Reply:
@getdave but I think this is expected with the current `mapping` object in the test file, because the default mapping is 1:1, therefore `nofollow` and `sponsored` linked with `linkRel` while `opensInNewTab` is not, so the expected `linkRel` is correct 'nofollow sponsored'